core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ProducerBindingService.java

```diff
@@ -79,14 +80,25 @@ public void delete(ProducerBinding producerBinding) {
   }
 
   public Iterable<ProducerBinding> findAll(Predicate<ProducerBinding> filter) {
-    return stream(producerBindingRepository.findAll().spliterator(), false)
-        .filter(r -> filter.test(r))
-        .collect(Collectors.toList());
+    return producerBindingRepository.findAll().stream().filter(filter).collect(toList());
+  }
+
+  public Optional<ProducerBinding> find(ProducerKey key) {
+    ProducerBinding example = new ProducerBinding(
+        new ProducerBindingKey(
+            key.getStreamDomain(),
+            key.getStreamName(),
+            key.getStreamVersion(),
+            key.getZone(),
+            null,
+            key.getName()
+        ), null, null);
+    return producerBindingRepository.findAll(Example.of(example)).stream().findFirst();
```

Comment:
This is where the magic happens.

Reply:
So this produces an actual HQL/ SQL query which matches the example, right?

Reply:
I imagine so.